monitor/query.go

```diff
@@ -145,15 +157,25 @@ func (m *Monitor) queryEndedEpochBTCHeightWithRetry(epoch uint64) (*monitortypes
 func (m *Monitor) queryReportedCheckpointBTCHeightWithRetry(hashStr string) (*monitortypes.QueryReportedCheckpointBtcHeightResponse, error) {
 	var reportedCheckpointBtcHeightRes monitortypes.QueryReportedCheckpointBtcHeightResponse
 
-	if err := retry.Do(m.ComCfg.RetrySleepTime, m.ComCfg.MaxRetrySleepTime, func() error {
+	retryOptions := []retry.Option{
+		retry.Delay(m.ComCfg.RetrySleepTime),
+		retry.MaxDelay(m.ComCfg.MaxRetrySleepTime),
+	}
+
+	if m.ComCfg.TestMode {
+		retryOptions = append(retryOptions, retry.Attempts(1))
```

Comment:
ensure that functions in  `ReportedCheckpointBTCHeight` in FuzzLivenessChecker execute only once

Reply:
maybe we could add `attempts` to config and set it to 1  in tests  ? Having special testmode in config is imo too hacky 😅